From e91ebc6e5e8993a08006b77e9b0e6c448562c982 Mon Sep 17 00:00:00 2001
From: Johan Kiviniemi <devel@johan.kiviniemi.name>
Date: Tue, 31 Aug 2010 01:08:15 +0300
Subject: Makefile.linux: Add -g to CFLAGS and -Wl,--as-needed to LDFLAGS

---
 Makefile.linux |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/Makefile.linux b/Makefile.linux
index 28c3b18..c4bd65d 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -26,6 +26,9 @@ STRIP ?= strip
 CP = cp
 
 
+CFLAGS += -g
+LDFLAGS += -Wl,--as-needed
+
 INSTALLPREFIX ?= $(PREFIX)/usr
 SBININSTALLDIR = $(INSTALLPREFIX)/sbin
 ETCINSTALLDIR = $(PREFIX)/etc/miniupnpd
-- 
1.7.4.1

