From 022afb4ddfb564f22c8b01975be1bb49503068b0 Mon Sep 17 00:00:00 2001
From: Johan Kiviniemi <devel@johan.kiviniemi.name>
Date: Tue, 31 Aug 2010 01:08:15 +0300
Subject: Makefile.linux: Add -g to CFLAGS and -Wl,--as-needed to LDFLAGS

---
 Makefile.linux |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/Makefile.linux b/Makefile.linux
index dfd0546..fdb6602 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -23,6 +23,9 @@ RM = rm -f
 INSTALL = install
 STRIP = strip
 
+CFLAGS += -g
+LDFLAGS += -Wl,--as-needed
+
 INSTALLPREFIX ?= $(PREFIX)/usr
 SBININSTALLDIR = $(INSTALLPREFIX)/sbin
 ETCINSTALLDIR = $(PREFIX)/etc/miniupnpd
-- 
1.6.3.3

