From 0fd6e3b21ba545c3c1867a0fa08b287cb9e4d28b Mon Sep 17 00:00:00 2001
From: Johan Kiviniemi <devel@johan.kiviniemi.name>
Date: Tue, 31 Aug 2010 01:10:24 +0300
Subject: Makefile.linux: Set libiptc-related flags via pkg-config

---
 Makefile.linux |   53 +++++++++++++++++++++--------------------------------
 1 files changed, 21 insertions(+), 32 deletions(-)

diff --git a/Makefile.linux b/Makefile.linux
index 9a025b5..0b9796d 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -41,38 +41,27 @@ NETFILTEROBJS = netfilter/iptcrdr.o
 
 ALLOBJS = $(BASEOBJS) $(LNXOBJS) $(NETFILTEROBJS)
 
-LIBS = -liptc
-# the following is better, at least on gentoo with iptables 1.4.6
-# see http://miniupnp.tuxfamily.org/forum/viewtopic.php?p=1618
-#LIBS = -lip4tc
-
-ifdef IPTABLESPATH
-CFLAGS := $(CFLAGS) -I$(IPTABLESPATH)/include/
-LDFLAGS := $(LDFLAFGS) -L$(IPTABLESPATH)/libiptc/
-# get iptables version and set IPTABLES_143 macro if needed
-IPTABLESVERSION := $(shell grep "\#define VERSION" $(IPTABLESPATH)/config.h | tr -d \" |cut -d" " -f3 )
-IPTABLESVERSION1 := $(shell echo $(IPTABLESVERSION) | cut -d. -f1 )
-IPTABLESVERSION2 := $(shell echo $(IPTABLESVERSION) | cut -d. -f2 )
-IPTABLESVERSION3 := $(shell echo $(IPTABLESVERSION) | cut -d. -f3 )
-# test if iptables version >= 1.4.3
-TEST := $(shell [ \( \( $(IPTABLESVERSION1) -ge 1 \) -a \( $(IPTABLESVERSION2) -ge 4 \) \) -a \( $(IPTABLESVERSION3) -ge 3 \) ] && echo 1 )
-ifeq ($(TEST), 1)
-CFLAGS := $(CFLAGS) -DIPTABLES_143
-# the following sucks, but works
-#LIBS = $(IPTABLESPATH)/libiptc/.libs/libip4tc.o
-LIBS = $(IPTABLESPATH)/libiptc/.libs/libiptc.a
-else
-LIBS = $(IPTABLESPATH)/libiptc/libiptc.a
-endif
-else
-# check for system-wide iptables files. Test if iptables version >= 1.4.3
-TEST := $(shell test -f /usr/include/iptables/internal.h && grep -q "\#define IPTABLES_VERSION" /usr/include/iptables/internal.h && echo 1)
-ifeq ($(TEST), 1)
-CFLAGS := $(CFLAGS) -DIPTABLES_143
-#LIBS = /usr/lib/libiptc.a
-LIBS = -liptc
-endif 
-endif
+CFLAGS := $(CFLAGS) $(shell pkg-config --cflags libiptc)
+LDLIBS := $(LDLIBS) $(shell pkg-config --libs libiptc)
+CFLAGS := $(CFLAGS) $(shell \
+	ver() { \
+		local old_IFS="$$IFS"; \
+		IFS=.; \
+		set -- $$1; \
+		IFS="$$old_IFS"; \
+		for arg; do \
+			if ! [ "$$arg" -ge 0 ]; then \
+				printf '%s\n' 0; \
+				return; \
+			fi; \
+		done; \
+		printf '%s\n' "$$((1000000*$${1:-0}+1000*$${2:-0}+$${3:-0}))"; \
+	}; \
+	iptc_ver="$$(ver "$$(pkg-config --modversion libiptc)")"; \
+	if [ "$$iptc_ver" -ge "$$(ver 1.4.3)" ]; then \
+		printf '%s\n' -DIPTABLES_143; \
+	fi; \
+)
 
 TESTUPNPDESCGENOBJS = testupnpdescgen.o upnpdescgen.o
 
-- 
1.6.3.3

