From 70a2f0cf7c22f330e68d74e3f5ea3ffe0d133792 Mon Sep 17 00:00:00 2001
From: Johan Kiviniemi <devel@johan.kiviniemi.name>
Date: Tue, 31 Aug 2010 01:10:24 +0300
Subject: Makefile.linux: Set libiptc-related flags via pkg-config

---
 Makefile.linux |   76 +++++++++++++++++---------------------------------------
 1 files changed, 23 insertions(+), 53 deletions(-)

diff --git a/Makefile.linux b/Makefile.linux
index c4bd65d..37d5b55 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -44,59 +44,29 @@ NETFILTEROBJS = netfilter/iptcrdr.o
 
 ALLOBJS = $(BASEOBJS) $(LNXOBJS) $(NETFILTEROBJS)
 
-ifeq "$(wildcard /etc/gentoo-release )" ""
-LIBS ?= -liptc
-else # gentoo
-# the following is better, at least on gentoo with iptables 1.4.6
-# see http://miniupnp.tuxfamily.org/forum/viewtopic.php?p=1618
-# and http://miniupnp.tuxfamily.org/forum/viewtopic.php?p=2183
-LIBS ?= -lip4tc
-CFLAGS := -DIPTABLES_143 $(CFLAGS)
-endif
-
-ARCH ?= $(shell uname -m | grep -q "x86_64" && echo 64)
-ifdef IPTABLESPATH
-CFLAGS := $(CFLAGS) -I$(IPTABLESPATH)/include/
-LDFLAGS := $(LDFLAFGS) -L$(IPTABLESPATH)/libiptc/
-# get iptables version and set IPTABLES_143 macro if needed
-ifeq ($(TARGET_OPENWRT),)
-IPTABLESVERSION := $(shell grep "\#define VERSION" $(IPTABLESPATH)/config.h | tr -d \" |cut -d" " -f3 )
-IPTABLESVERSION1 := $(shell echo $(IPTABLESVERSION) | cut -d. -f1 )
-IPTABLESVERSION2 := $(shell echo $(IPTABLESVERSION) | cut -d. -f2 )
-IPTABLESVERSION3 := $(shell echo $(IPTABLESVERSION) | cut -d. -f3 )
-# test if iptables version >= 1.4.3
-TEST := $(shell [ \( \( $(IPTABLESVERSION1) -ge 1 \) -a \( $(IPTABLESVERSION2) -ge 4 \) \) -a \( $(IPTABLESVERSION3) -ge 3 \) ] && echo 1 )
-ifeq ($(TEST), 1)
-CFLAGS := $(CFLAGS) -DIPTABLES_143
-# the following sucks, but works
-LIBS = $(IPTABLESPATH)/libiptc/.libs/libip4tc.o
-#LIBS = $(IPTABLESPATH)/libiptc/.libs/libiptc.a
-else # ifeq ($(TEST), 1)
-LIBS = $(IPTABLESPATH)/libiptc/libiptc.a
-endif # ifeq ($(TEST), 1)
-else # ($(TARGET_OPENWRT),)
-# openWRT :
-# check for system-wide iptables files. Test if iptables version >= 1.4.3
-# the following test has to be verified :
-TEST := $(shell test -f /usr/include/iptables/internal.h && grep -q "\#define IPTABLES_VERSION" /usr/include/iptables/internal.h && echo 1)
-ifeq ($(TEST), 1)
-CFLAGS := $(CFLAGS) -DIPTABLES_143
-LIBS = -liptc
-endif  # ($(TEST), 1)
-TEST_LIB := $(shell test -f /usr/lib$(ARCH)/libiptc.a && echo 1)
-ifeq ($(TEST_LIB), 1)
-LIBS = -liptc /usr/lib$(ARCH)/libiptc.a
-endif # ($(TEST_LIB), 1)
-endif # ($(TARGET_OPENWRT),)
-else # ifdef IPTABLESPATH
-# IPTABLESPATH not defined
-# the following test has to be verified :
-TEST := $(shell test -f /usr/include/xtables.h && grep -q "XTABLES_VERSION_CODE" /usr/include/xtables.h && echo 1)
-ifeq ($(TEST), 1)
-CFLAGS := $(CFLAGS) -DIPTABLES_143
-LIBS = -liptc
-endif # ($(TEST), 1)
-endif # ifdef IPTABLESPATH
+CFLAGS := $(CFLAGS) $(shell pkg-config --cflags libiptc)
+# NOTE: --static seems to be needed because miniupnpd uses the private
+# libip4tc.so.
+LDLIBS := $(LDLIBS) $(shell pkg-config --static --libs libiptc)
+CFLAGS := $(CFLAGS) $(shell \
+	ver() { \
+		local old_IFS="$$IFS"; \
+		IFS=.; \
+		set -- $$1; \
+		IFS="$$old_IFS"; \
+		for arg; do \
+			if ! [ "$$arg" -ge 0 ]; then \
+				printf '%s\n' 0; \
+				return; \
+			fi; \
+		done; \
+		printf '%s\n' "$$((1000000*$${1:-0}+1000*$${2:-0}+$${3:-0}))"; \
+	}; \
+	iptc_ver="$$(ver "$$(pkg-config --modversion libiptc)")"; \
+	if [ "$$iptc_ver" -ge "$$(ver 1.4.3)" ]; then \
+		printf '%s\n' -DIPTABLES_143; \
+	fi; \
+)
 
 TESTUPNPDESCGENOBJS = testupnpdescgen.o upnpdescgen.o
 
-- 
1.7.4.1

