From d58aa960447c676a917e181b45839761dfe2e6e7 Mon Sep 17 00:00:00 2001
From: Johan Kiviniemi <devel@johan.kiviniemi.name>
Date: Tue, 31 Aug 2010 01:11:44 +0300
Subject: miniupnpd.c: Do not log sun_path, as it seems to contain random memory

---
 miniupnpd.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/miniupnpd.c b/miniupnpd.c
index 943ceea..2f64558 100644
--- a/miniupnpd.c
+++ b/miniupnpd.c
@@ -1142,7 +1142,9 @@ main(int argc, char * * argv)
 			//syslog(LOG_DEBUG, "sctl!");
 			s = accept(sctl, (struct sockaddr *)&clientname,
 			           &clientnamelen);
+			/* Disabled: sun_path seems to contain random memory.
 			syslog(LOG_DEBUG, "sctl! : '%s'", clientname.sun_path);
+			*/
 			tmp = malloc(sizeof(struct ctlelem));
 			tmp->socket = s;
 			LIST_INSERT_HEAD(&ctllisthead, tmp, entries);
-- 
1.6.3.3

